### PR TITLE
Change gasnet source directory structure.

### PR DESCRIPTION
**Pull request by gbtitus**
(This follows the lead of similar changes to third-party/jemalloc and
third-party/hwloc.)

Rename GASNet-1.26.0. => gasnet-src so that we can avoid adding and
removing new/old source versions when we upgrade.

We used to add the version number to our 3rd party sources.  When we
upgrade versions we would add/commit the new tarball and the untarred
source.  The we would remove the tarball, and switch our makefiles to
use the new version.  Finally, after enough testing, we would remove the
old sources.

While this was nice because it allowed us to switch between two versions
easily and embedded the version name in the directory structure, it
added a lot of bloat to the git history.  We had two commits to add and
remove the tarball, and instead of just committing the diff between
releases, we ended up completing adding the new source in a new dir, and
removing the old source later.  This new setup will allow us to just
commit the diff between releases, which should drastically reduce the
amount stored in the git history.

### CODE REVIEW

**Comment by gbtitus** [comment]
@ronawho , would you please review this one?

**Comment by ronawho** [comment]
Hard to tell if this is included since github can't show me the whole diff, but did you update references to GASNET_VERSION in doc/developer/bestPractices/GASNetOnDesktops.txt?

**Comment by ronawho** [comment]
And I missed this in my hwloc PR, but I think you also need to update third-party/README references to <version> to say src. i.e. `GASNet-<version>` => `GASNet-src`

**Comment by ronawho** [comment]
Comment at https://github.com/chapel-lang/chapel/blob/master/runtime/src/comm/gasnet/comm-gasnet.c#L641 also needs to be updated

**Comment by ronawho** [comment]
There's also couple of places that refer to `GASNet-*` that can be changed to `GASNet-src` in doc/release/platforms/cray.rst and doc/release/multilocale.rst.

I just did a case insensitive grep for "GASNET-" and "GASNET_" excluding the gasnet-src dir. The GASNET_ version turns up a lot of results for function calls, so it's possible I missed a few places

File: third-party/README.devel [hunk 1]
```diff
@@ -0,0 +1,41 @@
+=========================================
+Chapel third-party developer instructions
+=========================================
+
+For certain third-party packages, currently gasnet, hwloc, and jemalloc,
```
Comment (ronawho) [inline third-party/README.devel]: The list of current packages is very, very likely to become out of date, so I might just remove it, or say something more like "some examples include"

**Pull request merged**